packages/contracts/test/test-utils/proxy.ts

```diff
@@ -0,0 +1,29 @@
+import {ContractFactory} from 'ethers';
+import {upgrades} from 'hardhat';
+
+type DeployOptions = {
+  constructurArgs?: unknown[];
+  proxyType?: 'uups';
+};
+
+// Used to deploy the implementation with the ERC1967 Proxy behind it.
+// Designed this way as It might be desirable to avoid OpenZeppelin upgrades package
+// later on and this way, all will be needed is to change this function.
+// NOTE: To avoid lots of changes in the whole test codebase, deployWithProxy
+// won't automatically call initialize and it's the caller's responsibility to do so.
+export async function deployWithProxy(
```

Comment:
rewrite the function to take in a generic and return it:
`deployWithProxy<T>(....): T` so we don't have this pattern when we use the function `await deployWithProxy(MerkleDistributor) as MerkleDistributor;` and can replace it with this `await deployWithProxy<MerkleDistributor>(MerkleDistributor);`

Reply:
https://github.com/aragon/core/pull/154/commits/fffac2202953dbdf7bdd43d4c63d9f4f6e9daec8